test/common/stats/stat_merger_fuzz_test.cc

```diff
@@ -0,0 +1,65 @@
+#include <algorithm>
+
+#include "common/stats/stat_merger.h"
+
+#include "test/common/stats/stat_test_utility.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Stats {
+namespace Fuzz {
+
+void testDynamicEncoding(absl::string_view data, SymbolTable& symbol_table) {
+  StatNameDynamicPool dynamic_pool(symbol_table);
+  StatNamePool symbolic_pool(symbol_table);
+  std::vector<StatName> stat_names;
+
+  for (uint32_t index = 0; index < data.size();) {
```

Comment:
@asraa this is the block of code where I take fuzz input in the form of a single string, and use it to generate an array of strings. Maybe this might be generally useful?

Reply:
I think the `num_bytes` calculation and the construction of `segment` is encapsulated by the `FuzzedDataProvider::ConsumeBytes(num_bytes)` function. 

But I can see the value of creating containers populated with fuzzed data. Possibly `FuzzedDataProvider` would support it? 

Generally, I know that the direction of fuzzing is going in this direction -- there are proposals for letting fuzzers consume more complex data, with something like `DEFINE_FUZZER(std::vector<std::string> input)`.

Reply:
One advantage of this version is that it doesn't copy the bytes, if perf is a concern. It looks like the impls in that library copy all the bytes, for reasons I haven't been able to discern. Looks like there is backslash processing in there.

Reply:
Wow... you're right, that is very odd to me. This is the best way to achieve that then. I can leave some feedback about that (byte copying/container population) if you'd like. The copying this is very weird.